packages/app-mobile/components/screens/SearchScreen/index.tsx

```diff
@@ -53,11 +53,36 @@ const useStyles = (theme: ThemeStyle, visible: boolean) => {
 	}, [theme, visible]);
 };
 
+// Workaround for https://github.com/laurent22/joplin/issues/12823:
+// Disable search-as-you-type for short 0-2 character searches that
+// are likely to match the start of a large number of words.
+const useSearchPaused = (query: string) => {
+	const [pauseDisabled, setPauseDisabled] = useState(false);
+	// Only disable search-as-you-type for a subset of all characters.
+	// This is, for example, to ensure that search-as-you-type remains
+	// enabled for CJK characters (e.g. U+6570 has length 1).
+	const paused = query.match(/^[a-z0-9]{0,2}$/i);
```

Comment:
Note: To disable search-as-you-type for *all* length 0-2 strings, we could instead use:
```suggestion
	// This is, for example, to ensure that search-as-you-type remains
	// enabled for CJK characters (e.g. U+6570 has length 1).
	const paused = query.match(/^.{0,2}$/i);
```